### DISCUSSION TITLE
在使用WaterfallPanel时出现异常

### DISCUSSION THREAD

**Discussion by BFNOC** [HandyOrg/HandyControl#1108] [Q&A]
异常内容：
> System.InvalidOperationException:“元素“HandyControl.Controls.WaterfallPanel”的布局测量覆盖不应将 PositiveInfinity 作为其 DesiredSize 返回，即使将 Infinity 作为可用大小传入。”

我是在ItemsControl里面使用WaterfallPanel的

```xaml
<ItemsControl ItemsSource="{Binding Value}">
  <ItemsControl.ItemsPanel>
      <ItemsPanelTemplate>
          <hc:WaterfallPanel Groups="8"/>
      </ItemsPanelTemplate>
  </ItemsControl.ItemsPanel>
```

这个问题我在网上搜过几遍，没有找到解决办法，有大佬可以解决吗？

**Comment by Yopler** [comment]
Hi 🖐,
I just test the `WaterfallPanel` into a `ItemsPanelTemplate` and it's working perfectly for me. Maybe your exception is coming from your `ItemsSource` Bind : Value

Here my code :

MainWindow.xaml.cs
```cs
public partial class MainWindow : Window
    {
        public List<string> MyList { get; set; }

        public MainWindow()
        {
            InitializeComponent();
            DataContext = this;
            MyList = new List<string>() {"Test","Test", "Test", "Test", "Test", "Test", "Test", };
        }
}
```

MainWindow.xaml
```xaml
        <ItemsControl ItemsSource="{Binding MyList}">
            <ItemsControl.ItemsPanel>
                <ItemsPanelTemplate>
                    <hc:WaterfallPanel Groups="8"/>
                </ItemsPanelTemplate>
            </ItemsControl.ItemsPanel>
        </ItemsControl>
```

Result : 
![image](https://user-images.githubusercontent.com/45144437/149980568-0ee78996-ef51-4d24-ab03-921058e95bf7.png)
